sdks/go/examples/wasm/wasm.go

```diff
@@ -0,0 +1,134 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// wasm is a simple example that loads and executes a wasm file function.
+// simple.wasm and simple.rs were copied from the example provided by the wasmer-go library:
+// https://github.com/wasmerio/wasmer-go/tree/master/examples/appendices
+//
+// New Concepts:
+// 1. Load a wasm file compiled using https://wasmer.io/
+// 2. Execute a wasm function within a DoFn
+package main
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/register"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/beamx"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/x/debug"
+	"github.com/wasmerio/wasmer-go/wasmer"
```

Comment:
Tangent question: Did you consider Wazero instead? https://github.com/tetratelabs/wazero
Wasmer is robustly supported (and we should have this example), but wazero has the non-trivial benefit of no CGO dependencies. (Not that I see using beam outside of AMD64 and aarch64, but it means it's all quickly compiled with Go.)